libkbfs/block_retrieval_queue.go

```diff
@@ -173,7 +173,7 @@ func (brq *blockRetrievalQueue) Request(ctx context.Context, priority int, kmd K
 			// specific type where the request blocks are CommonBlocks, but
 			// that direction can Set correctly. The cache will never have
 			// CommonBlocks.
-			cachedBlock, hasPrefetched, err :=
```

Comment:
Since `lifetime` is passed into `Request()`, I didn't know what to do if `bcache.Get()` returns something different.  I _think_ it should be the case that all requests that end up hitting the block retrieval queue will be transient entries anyway, since permanent entries by definition are guaranteed to be in the cache and would have been returned earlier by `folderBlockOps.getBlockHelperLocked()`.

What race are you worried about exactly?  Something getting re-put as Permanent after it's transitioned to Transient?  Even if the above logic is wrong, I think we can fix that by not calling `PrefetchAfterBlockRetrieved` when `hasPrefetched == true` in the cache already, since there's no point.

Anyway, I'll add a TODO here and merge as is, but let me know if you want me to fix something in a follow up PR.